### PR TITLE
Fix bug in the LeveragingBaggingClassifier and NaiveBayes

### LINKED ISSUE #240: [BUG] Leveraging Bagging bug where input to drift detector is inverted.
**Describe the bug**
At line 287, the correct and incorrect classifications are reversed. At the moment, it is:
`add = 1 if (pred[0] == y_coded) else 0`
but should be:
`add = 0 if (pred[0] == y_coded) else 1`

In addition, while using Leverage Bagging with NaiveBayes base classifier, the reset of the classifier (line 305, `self.ensemble[i_max].reset()`) results in the deletion of class list and priors and the newly created (reset) classifier throws TypeError and returns 0 (line 375) while predicting. I have avoided this by calling partial_fit after reset, but not sure if this is the best practice.

### PR DESCRIPTION
**Pull request by jacobmontiel**
Changes proposed in this pull request:

- Correct the value passed to ADWIN in `LeveragingBaggingClassifier` to monitor performance, previously this value was inverted.
- Add protection against division by zero.
- Update test cases and improve coverage.
- Remove dead code in `LeveragingBaggingClassifier`.
- Replace `adwin_ensemble` creation with list comprehension.
- Remove obsolete `try` in `LeveragingBaggingClassifier ` from the time when not all models were defaulting to prediction=[0] when empty. This is no longer necessary as now all classifiers return a valid value.
- Make the output of `predict_proba` in `NaiveBayes` 2D when the model is empty.
- Fixes #240 

---

#### Implementation
- [x] Implementation is correct (it performs its intended function).
- [x] Code complies with PEP-8 and is consistent with the framework.
- [x] Code is properly documented.
- [x] PR description covers ALL the changes performed.
- [x] Files changed (update, add, delete) are in the PR's scope (no extra files are included).

#### Tests
- [ ] New functionality is tested.
- [x] Tests are created for the new functionality or existing tests are updated accordingly.
- [x] ALL tests pass with no errors.
- [x] CI/CD pipelines run with no errors.
- [x] Test Coverage is maintained (coverage may drop by no more than 0.2%).

### CODE REVIEW

File: src/skmultiflow/meta/leverage_bagging.py [hunk 1]
```diff
@@ -244,7 +244,8 @@ def __partial_fit(self, X, y):
                     k = 1.0
                 elif pred[0] != y:
                     k = 1.0
-                elif self._random_state.rand() < (error / (1.0 - error)):
+                elif (error != 1.0 and
+                      self._random_state.rand() < (error / (1.0 - error))):
```
Comment (smastelini) [inline src/skmultiflow/meta/leverage_bagging.py:248]: Could you elaborate on the reason for the additional check (`error != 1.0`)?

File: src/skmultiflow/meta/leverage_bagging.py [hunk 2]
```diff
@@ -281,17 +282,14 @@ def __partial_fit(self, X, y):
...
-                pred = self.ensemble[i].predict(np.asarray([X]))
-                if pred is not None:
-                    add = 1 if (pred[0] == y_coded) else 0
-                    error = self.adwin_ensemble[i].estimation
-                    self.adwin_ensemble[i].add_element(add)
-                    if self.adwin_ensemble[i].detected_change():
-                        if self.adwin_ensemble[i].estimation > error:
-                            change_detected = True
-            except ValueError:
-                change_detected = False
```
Comment (smastelini) [inline src/skmultiflow/meta/leverage_bagging.py:294]: Do not we expect this portion to throw a `ValueError` exception anymore? Could you elaborate on this change too?

File: src/skmultiflow/meta/leverage_bagging.py (hunk 1 above)
Comment (jacobmontiel) [reply src/skmultiflow/meta/leverage_bagging.py:248]: This part of the code was unprotected and resulted in a `DivisionByZeroError` when `error==1.`

File: src/skmultiflow/meta/leverage_bagging.py (hunk 2 above)
Comment (jacobmontiel) [reply src/skmultiflow/meta/leverage_bagging.py:294]: Sorry about that. This is an obsolete check from the time when not all models were defaulting to `prediction=[0]` when empty. This is no longer the case as now all `scikit-multiflow` classifiers return a valid value.

File: src/skmultiflow/meta/leverage_bagging.py (hunk 1 above)
Comment (smastelini) [reply src/skmultiflow/meta/leverage_bagging.py:246]: Got it! Thank you for the answer!

File: src/skmultiflow/meta/leverage_bagging.py (hunk 2 above)
Comment (smastelini) [reply src/skmultiflow/meta/leverage_bagging.py:294]: Thank you for the clarification!

**Pull request merged**